src/collection-view.js

```diff
@@ -101,7 +101,7 @@ Marionette.CollectionView = Marionette.View.extend({
   _onCollectionAdd: function(child, collection, opts) {
     var index;
     if (opts.at !== undefined) {
-      index = opts.at;
+      index = opts.index || opts.at;
```

Comment:
This is still a bug when using earlier versions of Backbone (and, it'll mess up when `opts.index` is 0).